zbar/processor.c

```diff
@@ -49,6 +53,86 @@ static inline int proc_open (zbar_processor_t *proc)
     return(_zbar_processor_open(proc, "zbar barcode reader", width, height));
 }
 
+#ifdef HAVE_DBUS
+void zbar_send_dbus(const char* sigvalue)
+{
+    DBusMessage* msg;
+    DBusMessageIter args;
+    DBusConnection* conn;
+    DBusError err;
+    int ret;
+    dbus_uint32_t serial = 0;
+
+    // initialise the error value
+    dbus_error_init(&err);
+
+    // connect to the DBUS system bus, and check for errors
+    conn = dbus_bus_get(DBUS_BUS_SYSTEM, &err);
+    if (dbus_error_is_set(&err)) {
+        fprintf(stderr, "Connection Error (%s)\n", err.message);
+        dbus_error_free(&err);
+    }
+    if (NULL == conn) {
+        exit(1);
```

Comment:
We are in a library. Could these `exit(1)` calls be return errors instead?

Reply:
I have it logging and returning now so that the dbus errors aren't fatal.